File: src/entities/hourly_rate/hourly_rate.cpp
#include <iostream>
#include "hourly_rate.hpp"

HourlyRate::HourlyRate(){}
HourlyRate::HourlyRate(int id) : id(id) {}
HourlyRate::HourlyRate(int id, float amount): id(id), amount(amount) {}

HourlyRate::~HourlyRate(){}


void HourlyRate::setId(int id){
    this->id = id;
}

void HourlyRate::setAmount(float amount){
    this->amount = amount;
}

int HourlyRate::getId() const{
    return id;
}

float HourlyRate::getAmount() const{
    return amount;
}

HourlyRate& HourlyRate::operator= (const HourlyRate& h1){
    id = h1.id;
    amount = h1.amount;
    return *this;
}


bool operator== ( const HourlyRate& h1, const HourlyRate& h2)
{
    return h1.id == h2.id;
}

std::ostream& operator<<(std::ostream& os, const HourlyRate& h)
{
    os << "HourlyRate - Id: " << h.id << ", Amount: " << h.amount << std::endl;
    return os;
}

HourlyRateException::HourlyRateException(const std::string & message):_message(message) {}
const char * HourlyRateException::what() const throw() {
      return _message.c_str();
}

File: uPark_client/entities/hourly_rate.py
class HourlyRate:
    def __init__(self, id = None, amount = None):
        self._id = id
        self._amount = amount


    def set_id(self, id):
        self._id = id

    def set_amount(self, amount):
        self._amount = amount


    def get_id(self):
        return self._id

    def get_amount(self):
        return self._amount

    def __eq__(self, other):
        if isinstance(other, HourlyRate):
            return self._id == other._id
        return NotImplemented

    def __str__(self):
        hourly_rate_info = (f"Hourly rate Id: {self._id}, amount: {self._amount}")
        return hourly_rate_info
